### PR TITLE
SONARIAC-879 Use Orchestrator JUnit5 extension

### PR DESCRIPTION
**Pull request by petertrr**
* Integrates newer version of Orchestrator, which is now pluggable via junit-jupiter extension.
* Switches integration tests from `junit-vintage` engine to the new `junit-jupiter`, because old engine's test suites were incompatible with new juniter extensions
* Excludes transitive dependency on `junit4` from `json-simple` and `sslr-testing-harness`. A transitive dependency of `sonar-analyzer-test-commons` can't be excluded, because `org.junit.Assert.assertEquals` is used in `SingleFileVerifier` from `sonar-analyzer-test-commons`, which we are using.

### CODE REVIEW

**Review by mstachniuk — APPROVED** [review]
LGTM, please see my comments and wait with merge after release

**Comment by mstachniuk** [comment]
The classes `KubernetesSeedProviderTest` & `TreeUtilsTest` still uses JUnit4 classes

**Comment by petertrr** [comment]
> The classes `KubernetesSeedProviderTest` & `TreeUtilsTest` still uses JUnit4 classes

Is `KubernetesSeedProviderTest` even used for anything? Seems like the import for `assertThat` has been incorrect

**Comment by mstachniuk** [comment]
Yes, there is no need to fix `KubernetesSeedProviderTest`.

File: its/plugin/src/test/java/org/sonarsource/iac/Tests.java [hunk 1]
```diff
@@ -19,16 +19,16 @@
...
+
 import java.io.File;
-import org.junit.ClassRule;
-import org.junit.runner.RunWith;
-import org.junit.runners.Suite;
 
-@RunWith(Suite.class)
-@Suite.SuiteClasses({
+@Suite
+@SelectClasses({
```
[the remark this answers is not in the document]
Comment (nils-werner-sonarsource) [reply its/plugin/src/test/java/org/sonarsource/iac/Tests.java:31]: I agree with @mstachniuk, the suite seems not necessary anymore.

**Pull request merged**